domain/tutoringModel/tyche.py

```diff
@@ -0,0 +1,303 @@
+import numpy as np
+
+from domain.tutoringModel.Tyche import tyche_config as t_config
+from domain.tutoringModel.Tyche import tyche_utils as t_utils
+from errors import errors as err
+
+
+class TycheAlgorithm:
+    """This algorithm calculates the learning path based on the\
+        approach by Staufer et al. and is called Tyche algorithm.
+    """
+
+    def __init__(
+        self,
+        learning_elements=None,
+    ):
+        self.learning_elements = learning_elements
+        self.learning_style = {}
+        self.init_probabilities = []
+        self.probabilities = []
+        self.learningpath = []
+        self.last_le = False
+
+    def RGB_preprocessing(self):
+        """converts the dictionary learning element
+        into a list with only the short name of the LE"""
+        classification_learning_element = ["none"] * 12
+
+        for le in self.learning_elements:

```

Comment:
maybe use mapping as suggested in tyche_utils to reduce complexity and improve clarity.

Reply:
done